### ISSUE TITLE
Supply scratch directory through tar_resources()

### ISSUE THREAD

**Issue by wlandau** [ropensci/targets#893]
## Prework

* [x] Read and agree to the [code of conduct](https://ropensci.org/code-of-conduct/) and [contributing guidelines](https://github.com/ropensci/targets/blob/main/CONTRIBUTING.md).
* [x] If there is [already a relevant issue](https://github.com/ropensci/targets/issues), whether open or closed, comment on the existing thread instead of posting a new issue.
* [x] New features take time and effort to create, and they take even more effort to maintain. So if the purpose of the feature is to resolve a struggle you are encountering personally, please consider first posting a "trouble" or "other" issue so we can discuss your use case and search for existing solutions first.
* [x] Format your code according to the [tidyverse style guide](https://style.tidyverse.org/).

## Proposal

Currently, the scratch directory is always `file.path(tar_config_get("store"), "scratch")`. This is and #891 may contribute to #889. It would be better to supply scratch through `tar_resources()` (after #892 is implemented). This also makes it easier to make sure the internals of each target object know about a good scratch directory.

**Comment by wlandau** [comment]
Deciding against this because it gets really messy. Local targets can use the local data store for scratch files, and cloud targets can use files in `tempdir()`. All this is managed through `path_scratch_dir()` in `R/utils_path.R`.
